packages/context/src/binding.ts

```diff
@@ -257,6 +274,22 @@ export class Binding {
   }
 
   /**
+   * Override any existing configuration on the context for
+   * this given binding and provide the given values for injection instead
+   * Any bindings not defined in the given dependencies will be resolved from
+   * the binding's context.
+   * @param deps
+   */
+  options(deps: BindingDependencies): this {
+    // This context is currently unbound.
+    this.subcontext = new Context();
+    for (const key in deps) {
+      this.subcontext.bind(key).to(deps[key]);
```

Comment:
I wonder if need to use a namespace for the locally bound values. For example, `$.<key>`?

Do we want to support other binding providers such as Class/Provider/Factory?

Reply:
We can't namespace the keys. If we do, then it doesn't satisfy the injection target anymore.

Reply:
Well, we can still use `@inject('$.config')` to access the locally bound `config`.

Reply:
The keys that other developers will be defining in their plugins aren't going to be changing, so we shouldn't be mutating the keys that users are setting overrides for.

If you're saying `.getValue` should check for namespaced overrides, I'm still confused as to how that would work. If the namespace is bound to a sub-context, then it doesn't really change anything. If the namespace is bound to the _same_ context, then what happens when you want to bind multiple keys to different overrides for the same resolved value? It's going to leave you with the same scenario as when you started.